routes/auth.routes.js

```diff
@@ -0,0 +1,194 @@
+        token: token,
+      });
+      user.save().then(() => {
+        req.logIn(user, () => {
+          res.redirect('/willkommen');
+        });
+      });
+    });
+});
+
+router.get('/verify-email-link/:token', (req, res) => {
+  if (req.user.token === req.params.token) {
+    req.user.verifiedEmail = true;
+    req.user.save().then(() => {
+      // more professional : res.redirect and set a flash message before
+      req.flash(
+        'message',
+        'Super! Du hast deine E-Mail erfolgreich verifiziert.'
+      );
+      res.redirect('/willkommen');
+    });
+  }
+});
+
+// willkommen-seite
+router.get('/willkommen', (req, res) => {
+  const messages = req.flash('message');
+  const verifiedEmail = req.user.verifiedEmail;
+
+  Recipe.find().then((data) => {
+    const filteredData = data.filter((recipe) => {
+      if (req.query.level && req.query.dishType && req.query.nutrition) {
+        return (
+          recipe.level === req.query.level &&
+          recipe.dishType === req.query.dishType &&
+          recipe.nutrition === req.query.nutrition
+        );
+      } else if (req.query.dishType && req.query.nutrition) {
+        return (
+          recipe.dishType === req.query.dishType &&
+          recipe.nutrition === req.query.nutrition
+        );
+      } else if (req.query.nutrition && req.query.level) {
+        return (
+          recipe.nutrition === req.query.nutrition &&
+          recipe.level === req.query.level
+        );
+      } else if (req.query.level && req.query.dishType) {
+        return (
+          recipe.level === req.query.level &&
+          recipe.dishType === req.query.dishType
+        );
+      } else if (req.query.level) {
+        return recipe.level === req.query.level;
+      } else if (req.query.nutrition) {
+        return recipe.nutrition === req.query.nutrition;
+      } else if (req.query.dishType) {
+        return recipe.dishType === req.query.dishType;
+      } else {
+        return recipe;
+      }
+    });
+
+    let recipeCounter = filteredData.length;
+
+    let uniqueNutrition = [];
+    data.forEach((recipe) => {
+      if (!uniqueNutrition.includes(recipe.nutrition)) {
+        uniqueNutrition.push(recipe.nutrition);
+      }
+    });
+
+    let uniqueDishtype = [];
+    data.forEach((recipe) => {
+      if (uniqueDishtype.includes(recipe.dishType) == false) {
+        uniqueDishtype.push(recipe.dishType);
+      }
+    });
+
+    let sortedRecipes = [...filteredData].sort((a,b) => { if (a.createdAt < b.createdAt) { return 1 } else { return -1 } }).slice(0,5)
+
+
+
+    console.log(sortedRecipes)
+    // let sortedRecipes = Recipe.find().then((data) =>  {
+    // }
+    // )
+
+    res.render('auth/personalized-page', {
+      user: req.user,
+      messages: messages,
+      verifiedEmail: verifiedEmail,
+      recipes: filteredData,
+      count: recipeCounter,
+      nutrition: uniqueNutrition,
+      dishType: uniqueDishtype,
+      sortedRecipe1: sortedRecipes[0],
```

Comment:
this you should probably pass as an array of 5 elements instead of 5 different variables